proposals/params-collections.md

```diff
@@ -47,28 +47,30 @@ A *parameter_collection* consists of an optional set of *attributes*, a `params`
 a *type*, and an *identifier*. A parameter collection declares a single parameter of the given type with the given name.
 The *type* of a parameter collection shall be one of the following valid target types for a collection expression
 (see https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#conversions):
-- A single dimensional *array type* `T[]`
+- A single dimensional *array type* `T[]`, in which case the *element type* is `T`
 - A *span type*
   - `System.Span<T>`
-  - `System.ReadOnlySpan<T>`
-- A *type* with a *[create method](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#create-methods)*,
-  which is at least as accessible as the declaring member, and with an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/draft-v9/standard/statements.md#1395-the-foreach-statement)
-  determined from a `GetEnumerator` instance method or enumerable interface, not from an extension method.
+  - `System.ReadOnlySpan<T>`  
+  in which cases the *element type* is `T`
+- A *type* with an appropriate *[create method](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#create-methods)*,
+  which is at least as accessible as the declaring member and a corresponding *element type* resulting from that determination
 - A *struct* or *class type* that implements `System.Collections.IEnumerable` where:
   - The *type* has a constructor that can be invoked with no arguments, and the constructor is at least as accessible as the declaring member.
   - The *type* has an instance (not an extension) method `Add` that can be invoked with a single argument of
     the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/draft-v9/standard/statements.md#1395-the-foreach-statement),
-    and the method is at least as accessible as the declaring member.
+    and the method is at least as accessible as the declaring member.  
+  in which case the *element type* is the *iteration type*
 - An *interface type*
   - `System.Collections.Generic.IEnumerable<T>`,
   - `System.Collections.Generic.IReadOnlyCollection<T>`,
   - `System.Collections.Generic.IReadOnlyList<T>`,
   - `System.Collections.Generic.ICollection<T>`,
-  - `System.Collections.Generic.IList<T>`
+  - `System.Collections.Generic.IList<T>`  
+  in which cases the *element type* is `T`
 
 In a method invocation, a parameter collection permits either a single argument of the given parameter type to be specified, or
-it permits zero or more arguments of the collection [iteration type](https://github.com/dotnet/csharpstandard/blob/draft-v9/standard/statements.md#1395-the-foreach-statement)
-to be specified. Parameter collections are described further in *[Parameter collections](#parameter-collections)*.
+it permits zero or more arguments of the collection's *element type* to be specified. 
```

Comment:
@AlekseyTs Shouldn't the arguments be implicitly convertible to the element type?

Reply:
> Shouldn't the arguments be implicitly convertible to the element type?

This section is about parameter declarations and what is valid to declare. It is not supposed to restate consumption rules. We didn't for param arrays and, I think, this statement is a verbatim copy of what we used to say for them.

Reply:
Basically I didn't change existing wording, unless there was a difference with param arrays that was worth mentioning.